src/main/java/Deadline.java

```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task {
+    private String by;
+
+    public Deadline(String description, String by) {
+        super(description, Task.TYPE_DEADLINE);
+        this.by = by;
+    }
+
+    /**
+     * Returns string in list entry format.
+     * eg. [T][ ] task1 (by: Sunday)
+     *
+     * @return The formatted string.
+     */
```

Comment:
I like the usage of Return and @param in the method documentation, however, perhaps it may not be necessary as this method is trivial. The method name could be improved(simply override toString?) instead to explain the method functionality alone.